hbase-asyncfs/src/main/java/org/apache/hadoop/hbase/io/asyncfs/FanOutOneBlockAsyncDFSOutput.java

```diff
@@ -354,10 +364,24 @@ private void setupReceiver(int timeoutMs) {
     }
   }
 
+  private static final String DUMMY_DFS_OUTPUT_STREAM_CLASS =
+    "org.apache.hadoop.hdfs.DummyDFSOutputStream";
+
+  private DFSOutputStream tryCreateDummyDFSOutputStream(EnumSet<CreateFlag> createFlags) {
+    try {
+      return (DFSOutputStream) Class.forName(DUMMY_DFS_OUTPUT_STREAM_CLASS).getConstructors()[0]
```

Comment:
Push this down to the helper, and check (and store the resolved class and constructor handle) if the class exists when initializing the helper just like we do for the other Hadoop-dependent cases ?

Should be faster.